Add example manifests

ulucinar opened the pull request:
<!--
Thank you for helping to improve Crossplane!

Please read through https://git.io/fj2m9 if this is your first time opening a
Crossplane pull request. Find us in https://slack.crossplane.io/messages/dev if
you need any help contributing.
-->

### Description of your changes

<!--
Briefly describe what this pull request does. Be sure to direct your reviewers'
attention to anything that needs special consideration.

We love pull requests that resolve an open Crossplane issue. If yours does, you
can uncomment the below line to indicate which issue your PR fixes, for example
"Fixes #500":

-->
In the context of: https://github.com/crossplane-contrib/terrajet/issues/48

Adds generated example manifests for the Terraform resources.

I have:

- [X] Read and followed Crossplane's [contribution process].
- [X] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

<!--
Before reviewers can be confident in the correctness of this pull request, it
needs to tested and shown to be correct. Briefly describe the testing that has
already been done or which is planned for this change.
-->

[contribution process]: https://git.io/fj2m9
An example generated manifest:
```yaml
# This example manifest is auto-generated, and has not been tested.
# Please make the necessary adjustments before using it.
#apiVersion: compute.gcp.jet.crossplane.io/v1alpha2
#kind: Instance
#metadata:
#  name: example
#spec:
#  forProvider:
#    bootDisk:
#    - initializeParams:
#      - image: debian-cloud/debian-9
#    machineType: e2-medium
#    metadata:
#      foo: bar
#    metadataStartupScript: echo hi > /test.txt
#    networkInterface:
#    - accessConfig:
#      - {}
#      networkRef:
#        name: example
#    scratchDisk:
#    - interface: SCSI
#    serviceAccount:
#    - email: ${google_service_account.default.email}
#      scopes:
#      - cloud-platform
#    tags:
#    - foo
#    - bar
#    zone: us-central1-a
#  providerConfigRef:
#    name: example
```

Corresponding Terraform [registry example](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance):
```hcl
resource "google_compute_instance" "default" {
  name         = "test"
  machine_type = "e2-medium"
  zone         = "us-central1-a"

  tags = ["foo", "bar"]

  boot_disk {
    initialize_params {
      image = "debian-cloud/debian-9"
    }
  }

  // Local SSD disk
  scratch_disk {
    interface = "SCSI"
  }

  network_interface {
    network = "default"

    access_config {
      // Ephemeral public IP
    }
  }

  metadata = {
    foo = "bar"
  }

  metadata_startup_script = "echo hi > /test.txt"

  service_account {
    # Google recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
    email  = google_service_account.default.email
    scopes = ["cloud-platform"]
  }
}
```

turkenh commented on examples-generated/accesscontextmanager/gcpuseraccessbinding.yaml:
```diff
@@ -0,0 +1,14 @@
+# This example manifest is auto-generated, and has not been tested.
+# Please make the necessary adjustments before using it.
+apiVersion: accesscontextmanager.gcp.jet.crossplane.io/v1alpha1
+kind: GcpUserAccessBinding
```
```
kind: GCPUserAccessBinding
```

Could we generate with the latest [acronyms in terrajet](https://github.com/crossplane/terrajet/blob/2c54e8491abae0f19919b7d61faeea5d994731b2/pkg/types/name/name.go#L137)?

ulucinar replied:
Done. As an example, please see `examples/generated/accesscontextmanager/gcpuseraccessbinding.yaml`.

turkenh reviewed (commented):
Thanks @ulucinar!
It is great to have generated examples for whole resources. Apart from the comment that I've added for using acronyms, I would like us to consider putting them in `generated` folder under `examples`, instead of `examples-generated` at the repo root.

ulucinar commented:
> Thanks @ulucinar! It is great to have generated examples for all resources 👍
> 
> Apart from the comment that I've added for using acronyms, I would like us to consider putting them in `generated` folder under `examples`, instead of `examples-generated` at the repo root.

Thank you @turkenh. I've moved the generated examples under to `examples/generated` as you suggested.

turkenh reviewed (approved):
Thanks @ulucinar!
I personally wouldn't prefer to comment out all yaml given it is already mentioned to be generated and needs to be tweaked, but not a blocker.

ulucinar commented:
Thanks @turkenh!
